src/DotNetProjectFile.Analyzers/Extensions/System.IO.FileSystemInfo.cs

```diff
@@ -11,4 +13,10 @@ public static IEnumerable<DirectoryInfo> GetAncestors(this FileInfo file)
             current = current.Parent;
         }
     }
+
+    public static FileInfo SelectFile(this DirectoryInfo directory, string path)
```

Comment:
Why are SelectFile and SelectDirectory exactly the same?

Reply:
Because that is how it (should) work: you have a path, you extend that path, and it is either a file, or a directory. And that is up to developer who uses the method.